test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DisasterRecovery/BackupServiceTests.cs

```diff
@@ -232,6 +232,44 @@ public void ScriptBackupWithAdvancedOptionsTest()
             testDb.Cleanup();
         }
 
+        /// <summary>
+        /// Test the correct script generation for different backup action types
+        /// </summary>
+        [Fact]
+        public void ScriptBackupWithDifferentActionTypesTest()
+        {
+            string databaseName = "testbackup_" + new Random().Next(10000000, 99999999);
+            SqlTestDb testDb = SqlTestDb.CreateNew(TestServerType.OnPrem, false, databaseName);
+            
+            // Create Full backup script
+            string script = GenerateScriptForBackupType(BackupType.Full, databaseName);
+
+            // Validate Full backup script 
+            Assert.Contains("BACKUP DATABASE", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("BACKUP LOG", script, StringComparison.OrdinalIgnoreCase);
+            Assert.DoesNotContain("DIFFERENTIAL", script, StringComparison.OrdinalIgnoreCase);
+            
+            // Create log backup script
+            script = GenerateScriptForBackupType(BackupType.TransactionLog, databaseName);
+            
+            // Validate Log backup script 
+            Assert.Contains("BACKUP LOG", script, StringComparison.OrdinalIgnoreCase);
```

Comment:
If this using the VS Assert? If so I'd suggest adding in a descriptive message saying what the problem is - the VS default messages are quite unhelpful most of the time

Reply:
Doesnt have a message parameter and gives good enough message so leaving as is.